### PR TITLE
[converter] Fix negative axis in quantize/dequantize lowering

### PR DESCRIPTION
**Pull request by eyupcanakman**
## Description:
- The coreai::quantize / coreai::dequantize lowering normalized a negative axis as axis + rank - 1, off by one from the eager op, which resolves it as axis + rank. A per-channel axis=-1 then landed one dimension early. When the channel and a neighbor dim share a size there is no shape error, so the converted model picks the wrong channel silently.
- Normalize a negative axis the same way the eager op does (axis + rank), so the converted program matches eager for every negative axis.

## Testing
- python unit test
- Added per-channel negative-axis numerical tests for quantize and dequantize that fail on the old formula and pass with the fix.

### CODE REVIEW

File: tests/ops/test_custom_ops.py [hunk 1]
```diff
@@ -411,6 +411,28 @@ def forward(self, x: Tensor) -> Tensor:
...
+                )
+                self.register_buffer("zero_point", torch.zeros(4, dtype=torch.int8))
+
+            def forward(self, x: Tensor) -> Tensor:
+                return torch.ops.coreai.quantize(
+                    x, self.scale, torch.int8, zero_point=self.zero_point, axis=-1
+                )
+
+        model = Model()
+        x = torch.randn(2, 4, 4)
```
Comment (pkmandke) [inline tests/ops/test_custom_ops.py]: Maybe we can use a different dim value at [1], so something like `(2,3,4)` for clarity on which axis is picked via the negative indexing.

File: tests/ops/test_custom_ops.py [hunk 2]
```diff
@@ -540,6 +562,28 @@ def forward(self, x: Tensor) -> Tensor:
...
+                )
+                self.register_buffer("zero_point", torch.zeros(4, dtype=torch.int8))
+
+            def forward(self, x: Tensor) -> Tensor:
+                return torch.ops.coreai.dequantize(
+                    x, self.scale, zero_point=self.zero_point, axis=-1
+                )
+
+        model = Model()
+        x = torch.randint(-128, 127, (2, 4, 4), dtype=torch.int8)
```
Comment (pkmandke) [inline tests/ops/test_custom_ops.py]: same comment as above

File: tests/ops/test_custom_ops.py (hunk 1 above)
Comment (eyupcanakman) [reply tests/ops/test_custom_ops.py]: Added (2, 3, 4) as you suggested and kept (2, 4, 4) too. Equal dims are the case where a wrong axis is silent instead of a reshape error, so both stay covered.

File: tests/ops/test_custom_ops.py (hunk 2 above)
Comment (eyupcanakman) [reply tests/ops/test_custom_ops.py]: Same here.

**Comment by jakesabathia2** [comment]
Thank you for the fix @eyupcanakman , can you please rebase on the latest main and re-push?

**Review by pkmandke — APPROVED** [review]
Lgtm!
@yixingli-apple can you please take a look as well?

**Review by yixingli-apple — APPROVED** [review]
